pkg/controller/masteruserrecord/sync.go

```diff
@@ -205,23 +206,28 @@ func (s *Synchronizer) openShift3XConsoleURL(apiEndpoint string) (string, error)
 	return url, nil
 }
 
-// alignReadiness checks if all embedded SAs are ready
-func (s *Synchronizer) alignReadiness() {
+// alignDisabled updates the status to Disabled if all all the embedded UserAccounts have Disabled status
+func (s *Synchronizer) alignDisabled() {
 	for _, uaStatus := range s.record.Status.UserAccounts {
-		if !isReady(uaStatus.Conditions) {
+		if !condition.HasConditionReason(uaStatus.Conditions, toolchainv1alpha1.ConditionReady, toolchainv1alpha1.UserAccountDisabledReason) {
 			return
 		}
 	}
-	s.record.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(s.record.Status.Conditions, toBeProvisioned())
+
+	if len(s.record.Status.Conditions) > 0 {
```

Comment:
@MatousJobanek this was a PR suggestion which i thought made sense to be on the safe side: https://github.com/codeready-toolchain/host-operator/pull/149#discussion_r384135230

Reply:
ok, could you please explain to me why @tinakurian? I don't really see any purpose or reason for having this check there. Why should we check the length of the conditions in MUR.status - it doesn't have any impact on the result. In addition, if there is no condition, then we should add a new one containing the disabled reason, which is exactly what will do the next line `condition.AddOrUpdateStatusConditions`

Reply:
It was me who asked to add this check. But I made a type in the proposed snippet.
The app idea was to check `s.record.Status.UserAccounts` (not `s.record.Status.Conditions`). So we wouldn’t set the mur status to disabled if there is no embedded user accounts/user account statues (== status is unknown).
Such a check would make the alignDusabled function more robust (set mur to disabled only if all (if any) the embedded ua statues are disabled). But in the context of the whole sync process it would not make any difference because by the time this function is called there is always at least one embedded status in the mur.
So I’m ok with both options:
1. Remove this check
2. Fix the check, which may help in the future if we refactor the code which calls this function.

Reply:
> The idea was to check s.record.Status.UserAccounts (not s.record.Status.Conditions). So we wouldn’t set the mur status to disabled if there is no embedded user accounts/user account statues (== status is unknown).

yes, that makes sense

> Fix the check, which may help in the future if we refactor the code which calls this function.

+1